terraform/gitlab/gitlab.tf.json.template.py

```diff
@@ -696,13 +696,17 @@ def qq(*words):
         },
         'aws_subnet': {
             # A public and a private subnet per availability zone
-            f'gitlab_{vpc.subnet_name(public)}_{zone}': {
+            f'{component}_{vpc.subnet_name(public)}_{zone}': {
                 'availability_zone': f'${{data.aws_availability_zones.available.names[{zone}]}}',
-                'cidr_block': f'${{cidrsubnet(aws_vpc.gitlab.cidr_block, 8, {vpc.subnet_number(zone, public)})}}',
+                'cidr_block': f'${{cidrsubnet(aws_vpc.gitlab.cidr_block, 8, {vpc.subnet_number(component, zone, public)})}}',
                 'map_public_ip_on_launch': public,
                 'vpc_id': '${aws_vpc.gitlab.id}'
             }
-            for public in (False, True)
+            for (component, public) in [
```

Comment:
I don't think you need to add a third subnet. The ES cluster should go into the existing private subnet.